src/main/java/org/spongepowered/api/event/cause/entity/damage/source/DamageSource.java

```diff
@@ -108,24 +108,99 @@ static Builder builder() {
      */
     boolean doesAffectCreative();
 
+    /**
+     * Gets the amount of exhaustion this {@link DamageSource} will
+     * add to the entity, generally only to players.
+     *
+     * <p>In vanilla this is generally set to 0.1 by default and
+     * overridden and set to 0 if the damage is set to be absolute.</p>
+     *
+     * @return The increase in exhaustion
+     */
+    double getExhaustion();
+
     interface Builder extends DamageSourceBuilder<DamageSource, Builder> { }
 
     interface DamageSourceBuilder<T extends DamageSource, B extends DamageSourceBuilder<T, B>> extends ResettableBuilder<T, B> {
 
+        /**
+         * Sets this {@link DamageSource}'s damage to be scaled
+         * by {@link Difficulty}.
+         *
+         * @return This builder
+         */
         B scalesWithDifficulty();
 
+        /**
+         * Sets this {@link DamageSource} as dealing damage that
+         * bypasses armor.
+         *
+         * @return This builder
+         */
         B bypassesArmor();
 
+        /**
+         * Sets whether this {@link DamageSource} as an explosion.
+         *
+         * @return This builder
+         */
         B explosion();
 
+
+        /**
+         * Sets this {@link DamageSource} as not being able to be modified
+         * with damage that is absolute.
```

Comment:
This is a bit confusing - do you actually mean that absolute damage cannot modify this?

What I imagine you mean is that this damage source will be unaffacted by damage modifiers, right?

Reply:
I copied this from the get method. I will clean both of them up, and look at how we exactly handle a damage source being absolute.

Reply:
Noted, may as well take advantage of this PR to do that, I guess.

Reply:
@dualspiral @gabizou So looking at it now(since we handle damage completely ourselves for the most part), it seems(not 100% sure) we only ignore armor and potion resistance if damage is absolute, but not enchantments, shields, and absorption. Is this an oversight and should we bypass those protections as well, or should I document absolute as just bypassing armor and potion resistance?

Reply:
Looks like we were ignoring armor if it was absolute instead of if it was bypassing armor, and were not bypassing enchantments if it was absolute. I will make these changes in my SpongeCommon PR.

To summarize: If a damage source is absolute, like the mcp Javadocs say, it seems to just bypass potions and enchantments.